threads/cpuinfo.h

```diff
@@ -81,8 +83,12 @@ namespace HL {
 class CPUInfo {
 public:
 
+  #ifdef HL_APPLE_SILICON
+  enum { PageSize = 16384UL };
```

Comment:
If `vm_page_size` is const, we should be able to use it here.

Reply:
Unfortunately, it appears to be a runtime-only value (cf. https://opensource.apple.com/source/xnu/xnu-4570.71.2/libsyscall/mach/mach/vm_page_size.h.auto.html). Quoting from the "Determining CPU page size" section of https://www.wwdcnotes.com/notes/wwdc20/10214/:

>  Native page size on Intel is 4 kB, on Apple Silicon it's 16 kB: therefore the PAGE_SIZE macro is no longer a constant. Use:
> - PAGE_MAX_SIZE for a compile-time upper bound
> - vm_page_size to read the actual value at runtime
>
> Rosetta provides 4 KB pages when translating Intel processes

Furthermore, `PAGE_MAX_SIZE` is very conservative and returns `16K` even when compiling for x86, as I confirmed with this janky program:

```c
#include <mach/vm_page_size.h>
#include <mach/vm_param.h>
#include <stdio.h>
#include <stdlib.h>

int main() {
    printf("PAGE_MAX_SIZE: %d\n", PAGE_MAX_SIZE);
    printf("vm_page_size: %lu\n", vm_page_size);
    printf("arch: ");
    fflush(stdout);
    system("uname -p");
    return 0;
}
```

Then I ran this on my M1 (it runs the x86 program in Rosetta):

```console
$ clang check_page_size.c -o check_page_size_arm
$ clang check_page_size.c -target x86_64-apple-darwin -o check_page_size_x86
$ ./check_page_size_arm
PAGE_MAX_SIZE: 16384
vm_page_size: 16384
arch: arm
$ ./check_page_size_x86
PAGE_MAX_SIZE: 16384
vm_page_size: 4096
arch: i386
```

So it appears we cannot use `PAGE_MAX_SIZE` since it's an upper-bound, and we can't use `vm_page_size` because it's runtime-only.